src/main/kotlin/org/rust/lang/core/types/RustTypeParameterType.kt

```diff
@@ -16,6 +16,8 @@ class RustTypeParameterType(val parameter: RustTypeParamElement) : RustTypeBase(
     override fun getNonStaticMethodsIn(project: Project): Sequence<RustFnElement> =
         getTraitsImplementedIn(project).flatMap { it.traitMethodMemberList.asSequence() }
 
+    override fun substitute(map: Map<RustTypeParameterType, RustType>): RustType = map[this] ?: this
+
```

Comment:
B/c you don't map _type_ into _type_, you map _parameter_ (which has nothing to do with types) into type.